lib/http/client.rb

```diff
@@ -123,7 +123,7 @@ def make_request_uri(uri, opts)
       uri = HTTP::URI.parse uri
 
       if opts.params && !opts.params.empty?
-        uri.query = [uri.query, HTTP::URI.form_encode(opts.params)].compact.join("&")
+        uri.query = [uri.query, HTTP::URI.encode_query(opts.params)].compact.join("&")
```

Comment:
I think we should expose `:query_values` method on `HTTP::URI` (delegate it to underlying `@uri`), this in here we will be able to:

``` ruby
uri.query_values = uri.query_values(Array).concat(opts.params.to_a)
```

Thus there will be no need in exposing `encode_query`.